src/core_scan/Core_scan.ml

```diff
@@ -891,16 +893,50 @@ let scan ?match_hook config ((rules, invalid_rules), rules_parse_time) :
         Core_profiling.Debug { skipped_targets; profiling }
     | (Core_profiling.Time _ | Core_profiling.No_info) as x -> x
   in
-
   { res with matches; errors; extra }
 
+(*****************************************************************************)
+(* Pre and Post Processors Hook For Semgrep Pro / Extensions        *)
+(*****************************************************************************)
+
+module type Pre_and_post_processor = sig
+  type state
+
+  val pre_process : Rule.t list -> Rule.t list * state
```

Comment:
This can be in the Pre_and_post_processor.mli, and it would be useful to add more comments on what those functions are doing.